lib/rules/require-computed-macros.js

```diff
@@ -115,6 +115,17 @@ function getThisExpressions(nodeLogicalExpression) {
   return arrayOfThisExpressions.reverse();
 }
 
+function detectCircularKey(node, macroArgs) {
+  if (node.parent && node.parent.key) {
```

Comment:
There should always be a parent, but we do need to check what node type the parent is before we check its `key`. That way, we won't end up using the `key` of some unexpected kind of node.

Reply:
oddly I believe I encountered a no parent situation in our own test suite. Will double check that.

Reply:
ESLint adds the `parent` when it runs the rule test cases. But if you write separate unit tests for this function by itself, then it won't have a parent. So feel free to add the parent guard but you only need if you write separate unit tests for the function itself.